bin/reth/src/args/rpc_server_args.rs

```diff
@@ -227,7 +228,9 @@ impl RpcServerArgs {
         });
 
         // launch servers concurrently
-        Ok(futures::future::try_join(launch_rpc, launch_auth).await?)
+        let handles = Ok(futures::future::try_join(launch_rpc, launch_auth).await?);
+
+        conf.on_rpc_server_started(self, components, &node_modules, handles)
```

Comment:
we want to pass the handles as a wrapper type, and only call on_rpc_started if Ok